platform/state/platform_state.proto

```diff
@@ -0,0 +1,206 @@
+/**
+ * # PlatformState
+ * Messages that hold platform state in the network state.
+ *
+ * The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+ * "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+ * document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119)
+ * and clarified in [RFC8174](https://www.ietf.org/rfc/rfc8174).
+ */
+syntax = "proto3";
+
+package com.hedera.hapi.platform.state;
+
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+option java_package = "com.hedera.hapi.platform.state.legacy";
+// <<<pbj.java_package = "com.hedera.hapi.platform.state">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+import "basic_types.proto";
+import "timestamp.proto";
+
+/**
+ * The current state of platform consensus.<br/>
+ * This message stores the current consensus data for the platform
+ * in network state.
+ *
+ * The platform state SHALL represent the latest round's consensus.<br/>
+ * This data SHALL be used to ensure consistency and provide critical data for
+ * restart and reconnect.
+ */
+message PlatformState {
+    /**
+     * A consensus round.<br/>
+     * The round represented by this state.
+     * <p>
+     * This message SHALL represent the network state after handling of all
+     * transactions reaching consensus in all previous rounds.<br/>
+     * The first state (genesis state) SHALL have a round of 0.<br/>
+     * The first round to handle transactions SHALL be round `1`.
+     */
+    uint64 round = 1;
```

Comment:
This value is duplicated inside of `ConsensusSnapshot`. Any reason it needs to exist in both places?

Reply:
The reason is that Java counterpart also have the same fields. Do you know if they mean exactly the same thing (current round) and one of them can be removed in the context of `PlatformState`?

Reply:
@cody-littley @lpetrovic05 Could you guys weigh in here?

It looks to me like these are the same.

Reply:
These are the same thing. We should have this data in one place or the other, not both. I'm not strongly opinionated one way or another which should stay and which should go.

Reply:
Sounds good. I will remove it from `ConsensusSnapshot` then

Reply:
@imalygin  The value in the platform state gets populated from the `ConsensusSnapshot`, so it must be removed from the platform state, not the snapshot

Reply:
@lpetrovic05 okay, that's even better. I'll remove it from here.